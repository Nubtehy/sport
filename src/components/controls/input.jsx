import React from 'react';
import PropTypes from 'prop-types';
import { withStyle } from 'components/HOCs';

const displayName = 'Input';

const propTypes = {
  onClick: PropTypes.func,
  style: PropTypes.objectOf(PropTypes.string),
  children: PropTypes.string,
  name: PropTypes.string,
};

const defaultProps = {
  onClick: undefined,
  style: {},
  children: '',
  name: '',
};

export function InputBase({
  name, onChange, style, placeholder, value
}) {
  return (
    <input name={name} type="text" onChange={onChange} style={style} value={value} placeholder={placeholder}/>
  );
}

InputBase.displayName = displayName;
InputBase.propTypes = propTypes;
InputBase.defaultProps = defaultProps;

export const styleCreator = (theme, { accent }) => ({
  backgroundColor: accent ? theme.colorAccent : theme.colorDefault,
});

export default withStyle(InputBase, styleCreator);
