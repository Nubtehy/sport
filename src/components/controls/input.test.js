import React from 'react';
import { shallow } from 'enzyme';

import { InputBase } from './input';

const testProps = {
  onClick: jest.fn(),
  style: { background: 'blue' },
  name: 'add-form-input',
};

describe('Given an Input component', () => {
  describe('when the component is rendered', () => {
    let component;

    beforeEach(() => {
      component = shallow(<InputBase {...testProps} />);
    });

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
