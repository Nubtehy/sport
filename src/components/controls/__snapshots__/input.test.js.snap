// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an Input component when the component is rendered should match the snapshot 1`] = `
<input
  name="add-form-input"
  onChange={[Function]}
  placeholder=""
  style={
    Object {
      "background": "blue",
    }
  }
  type="text"
  value=""
/>
`;
